packages/codegen-ui-react/lib/react-component-render-helper.ts

```diff
@@ -324,19 +325,46 @@ export function getSyntaxKindToken(operator: RelationalOperator): BinaryOperator
   }
 }
 
-export function getConditionalOperandExpression(operand: string | number | boolean): Expression {
-  switch (typeof operand) {
+export function getConditionalOperandExpression(
+  operand: string | number | boolean,
+  operandType: string | undefined,
+): Expression {
+  if (typeof operand === 'string' && operandType && operandType !== 'string') {
+    return valueToCorrectLiteral(operand, operandType);
+  }
+  return valueToJsxLiteral(operand);
+}
+
+function valueToCorrectLiteral(value: any, valueType: string): PrimaryExpression {
+  try {
+    const typedVal = JSON.parse(value);
+    if (valueType === typeof typedVal) {
+      return valueToJsxLiteral(typedVal);
+    }
+    throw Error(`Parsed value ${value} and type ${valueType} mismatch`);
+  } catch (err) {
+    if (err instanceof SyntaxError) {
+      return factory.createStringLiteral(value);
+    }
+    throw err;
+  }
+}
+
+function valueToJsxLiteral(value: any): PrimaryExpression {
+  switch (typeof value) {
     case 'number':
-      return factory.createNumericLiteral(operand);
+      return factory.createNumericLiteral(value);
     case 'boolean':
-      return operand ? factory.createTrue() : factory.createFalse();
+      return value ? factory.createTrue() : factory.createFalse();
+    case 'string':
+      return factory.createStringLiteral(value);
     default:
-      return factory.createStringLiteral(operand);
```

Comment:
I think I'd prefer to leave this behavior as-is for now. Is there a need to change this?